### ISSUE TITLE
Notebook can generate so much output that it crashes the browser

### ISSUE THREAD

**Issue by icankeep** [jupyterlab/jupyterlab#12687]
I opened a notebook with a lot of output and it crashed the browser, so I want to set a limit on the number of lines of output displayed when the notebook is opened.

![image](https://user-images.githubusercontent.com/39434627/173293099-73148171-b696-4c64-9626-0c35020c8d07.png)

I set the `maxNumberOutputs` parameter but it didn't work

![image](https://user-images.githubusercontent.com/39434627/173293162-af5a961d-8dc6-4756-a456-582e989b0cee.png)

**Comment by mlucool** [comment]
`maxNumberOutputs` features limits number of mime renders. If you have many lines of stdout/stderr, I think you should try https://github.com/deshaw/jupyterlab-limit-output

**Comment by icankeep** [comment]
@mlucool Thank you, but it is not valid for pandas data.


![image](https://user-images.githubusercontent.com/39434627/173486935-c714ac3a-6c9a-48e8-b2ac-7844005e5ed9.png)

**Comment by mlucool** [comment]
Correct. I don't think either approach will prevent footguns from a single render (e.g. the pandas render).

**Comment by jweill-aws** [comment]
If your pandas output is too large to display, you will need to reduce your output. Alternatively, see if the work that @fcollonval is doing on virtual notebooks (#12554) can help you.

The `maxNumberOutputs` parameter is not relevant to this, although there is an underlying issue that JupyterLab can generate so much output that it crashes your browser.
